src/hooks/useOverlayManager.tsx

```diff
@@ -1,17 +1,21 @@
-import React, { useCallback, useContext, useState } from 'react'
+import React, { useCallback, useContext, useState, useRef } from 'react'
 import { disableBodyScroll, enableBodyScroll } from 'body-scroll-lock'
+import styled from '@emotion/styled'
 import { css, Global } from '@emotion/react'
+import { zIndex } from '@/shared/theme'
 
 type OverlayManagerContextValue = {
   overlayOpen: boolean
   setOverlayOpen: (value: boolean, scrollbarGap?: number) => void
+  overlayContainerRef: React.RefObject<HTMLDivElement>
 }
 const OverlayManagerContext = React.createContext<OverlayManagerContextValue | undefined>(undefined)
 OverlayManagerContext.displayName = 'OverlayManagerContext'
 
 export const OverlayManagerProvider: React.FC = ({ children }) => {
   const [overlayOpen, setOverlayOpen] = useState(false)
```

Comment:
I think we will need to adjust naming now since this state doesn't actually represent whether there's an overlay opened but rather whether the scroll lock has been applied and I think we need to clear those up. Let's rename this to `scrollLocked` and add `overlayContainerOpened` boolean (we can apply conditional styles based on that, look the other comment).

Let's also clear up the naming for functions. I've originally named the function returned from the hook `handleOverlayOpen` because overlay manager only reacted with applying scroll lock to overlay being opened somewhere else. So my search overlay was calling `handleOverlayOpen` to basically say "hey I've opened an overlay so handle that in whatever way you need to". However, we cannot do the same with `handleOverlayContainerOpen` - this function's purpose isn't to _react_ to overlay container being opened, it's purpose it to actually _open_ the overlay container.

I propose the following names:
```ts
type OverlayManagerContextValue = {
  scrollLocked: boolean
  setScrollLocked: (value: boolean, scrollbarGap?: number) => void
  overlayContainerOpened: boolean
  setOverlayContainerOpened: (value: boolean) => void
  overlayContainerRef: React.RefObject<HTMLDivElement>
}
```

And returned from the hook:
```ts
{
  lockScroll(),
  unlockScroll(),
  openOverlayContainer(),
  closeOverlayContainer(),
  overlayContainerRef
}
```